### PR TITLE
Add small helper method for building EventMetadata

### PR DESCRIPTION
**Pull request by laurence-bird**

### CODE REVIEW

File: modules/core/src/main/scala/com/ovoenergy/comms/model/EventMetadata.scala [hunk 1]
```diff
@@ -17,3 +18,22 @@ case class EventMetadata(
...
+object EventMetadata {
+  def fromMetadata(metadataV3: MetadataV3): EventMetadata = {
+    EventMetadata(
+      traceToken = metadataV3.traceToken,
+      eventId = UUID.randomUUID().toString,
+      createdAt = Instant.now()
+    )
+  }
+
+  def fromGenericMetadata(genericMetadataV3: GenericMetadataV3): EventMetadata = {
```
Comment (ZsoltBalvanyos) [inline modules/core/src/main/scala/com/ovoenergy/comms/model/EventMetadata.scala]: changed this from GenericMetadata to GenericMetadataV3

File: modules/core/src/main/scala/com/ovoenergy/comms/model/Feedback.scala [hunk 2]
```diff
@@ -10,6 +11,8 @@ case class Feedback(
     @AvroDoc("The ID of the customer who is the target of the communication") customer: Option[Customer],
     @AvroDoc("The status of the comm") status: FeedbackStatus,
     @AvroDoc("A detailed description for the status") description: Option[String],
+    @AvroDoc("Email specific feedback") emailFeedback: Option[EmailFeedback],
```
Comment (filosganga) [inline modules/core/src/main/scala/com/ovoenergy/comms/model/Feedback.scala]: should we call this member just `email` (leaving type `EmailFeedback`) as the container type is `Feedback`, IMHO there is no need to repeat feedback.

File: modules/core/src/main/scala/com/ovoenergy/comms/model/FeedbackStatus.scala [hunk 3]
```diff
@@ -13,12 +13,12 @@ sealed trait FeedbackStatus extends LoggableEvent {
...
-  case object Delivered  extends FeedbackStatus
-  case object Opened     extends FeedbackStatus
-  case object Interacted extends FeedbackStatus
-  case object Failed     extends FeedbackStatus
+  case object Scheduled          extends FeedbackStatus
+  case object Pending            extends FeedbackStatus
+  case object Delivered          extends FeedbackStatus
+  case object Failed             extends FeedbackStatus
+  case object Cancelled          extends FeedbackStatus
+  case object FailedCancellation extends FeedbackStatus
```
Comment (filosganga) [inline modules/core/src/main/scala/com/ovoenergy/comms/model/FeedbackStatus.scala]: Should not be `CancellationFailed` 🤔?

File: modules/core/src/main/scala/com/ovoenergy/comms/model/sms/EmailFeedback.scala [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package com.ovoenergy.comms.model.sms
+
+sealed trait EmailFeedback
+
+/*
+    Can't be in companion object due to https://github.com/scala/bug/issues/10330
```
Comment (filosganga) [inline modules/core/src/main/scala/com/ovoenergy/comms/model/sms/EmailFeedback.scala]: This is what you were talking about? :'(

File: modules/core/src/test/scala/com/ovoenergy/comms/model/Arbitraries.scala [hunk 5]
```diff
@@ -93,13 +93,24 @@ trait Arbitraries extends JavaTimeArbitraries {
...
-    status      <- arbitrary[FeedbackStatus]
-    description <- option(genNonEmptyString)
-    metadata    <- arbitrary[EventMetadata]
-  } yield Feedback(commId, customer, status, description, metadata))
+    commId        <- genNonEmptyString
+    customer      <- option(arbitrary[Customer])
+    status        <- arbitrary[FeedbackStatus]
+    description   <- option(genNonEmptyString)
+    metadata      <- arbitrary[EventMetadata]
+    emailFeedback <- arbitrary[Option[EmailFeedback]]
```
Comment (filosganga) [inline modules/core/src/test/scala/com/ovoenergy/comms/model/Arbitraries.scala]: Ideally, it should be present only if the `channel` is `Email` not a big deal tough.

File: modules/core/src/main/scala/com/ovoenergy/comms/model/FeedbackStatus.scala (hunk 3 above)
Comment (laurence-bird) [reply modules/core/src/main/scala/com/ovoenergy/comms/model/FeedbackStatus.scala]: not sure if it makes a difference either way, wasn't sure so just kept it consistent with: https://github.com/ovotech/comms-kafka-messages/blob/master/modules/core/src/main/scala/com/ovoenergy/comms/model/FailedCancellationV3.scala

File: modules/core/src/main/scala/com/ovoenergy/comms/model/Feedback.scala (hunk 2 above)
Comment (laurence-bird) [reply modules/core/src/main/scala/com/ovoenergy/comms/model/Feedback.scala]: sure will update

File: modules/core/src/main/scala/com/ovoenergy/comms/model/EventMetadata.scala [hunk 6]
```diff
@@ -17,3 +18,22 @@ case class EventMetadata(
     "eventId"    -> eventId
   )
 }
+
+object EventMetadata {
+  def fromMetadata(metadataV3: MetadataV3): EventMetadata = {
+    EventMetadata(
+      traceToken = metadataV3.traceToken,
+      eventId = UUID.randomUUID().toString,
```
[the remark this answers is not in the document]
Comment (laurence-bird) [reply modules/core/src/main/scala/com/ovoenergy/comms/model/EventMetadata.scala]: Take your point thats a good idea. Will update

File: modules/core/src/main/scala/com/ovoenergy/comms/model/FeedbackStatus.scala (hunk 3 above)
Comment (filosganga) [reply modules/core/src/main/scala/com/ovoenergy/comms/model/FeedbackStatus.scala]: No idea, I am fine with both.

**Comment by filosganga** [comment]
LGTM

**Pull request merged**